api/v1beta1/common_types.go

```diff
@@ -101,6 +101,79 @@ type BarbicanComponentTemplate struct {
 	NetworkAttachments []string `json:"networkAttachments,omitempty"`
 }
 
+// +kubebuilder:validation:Enum=simple_crypto;pkcs11
+// This SecretStore type is used by the EnabledSecretStores variable inside the specification.
+type SecretStore string
+
+// BarbicanPKCS11Template - Includes all common HSM properties
+type BarbicanPKCS11Template struct {
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default=false
+	HSMEnabled bool `json:"hsmEnabled"`
+
+        // +kubebuilder:validation:Optional
+	// Path to vendor's PKCS11 library
+	HSMLibraryPath string `json:"hsmLibraryPath"`
+
+        // +kubebuilder:validation:Optional
+	// +kubebuilder:default="12345678"
+        // Token serial number used to identify the token to be used. Required
+	// when the device has multiple tokens with the same label.
+	HSMTokenSerialNumber string `json:"hsmTokenSerialNumber"`
+
+        // +kubebuilder:validation:Optional
+	// Token label used to identify the token to be used. Required when
+	// token_serial_number is not specified.
+	HSMTokenLabel string `json:"hsmTokenLabel"`
+
+        // +kubebuilder:validation:Optional
+	// OpenShift secret storing the password to login to PKCS11 session
+	HSMLogin string `json:"hsmLogin"`
+
+        // +kubebuilder:validation:Optional
+	// Label to identify master KEK in the HSM (must not be the same as HMAC label)
+	HSMMKEKLabel string `json:"hsmMKEKLabel"`
+
+        // +kubebuilder:validation:Optional
+	// +kubebuilder:default=32
+	// Length in bytes of master KEK
+	HSMMKEKLength int `json:"hsmMKEKLength"`
+
+        // +kubebuilder:validation:Optional
+	// Label to identify HMAC key in the HSM (must not be the same as MKEK label)
+	HSMHMACLabel string `json:"hsmHMACLabel"`
+
+        // +kubebuilder:validation:Optional
+	// +kubebuilder:default=1
+	// HSM Slot ID that contains the token device to be used
+	HSMSlotId int `json:"hsmSlotId"`
+
+        // +kubebuilder:validation:Optional
+        // +kubebuilder:default=4
+        // +kubebuilder:validation:Maximum=7
+        // +kubebuilder:validation:Minimum=0
+        // Level of logging, where 0 means "no logging" and 7 means "debug".
+        HSMLoggingLevel int `json:"hsmLoggingLevel"`
+
+	// +kubebuilder:validation:Optional
+	// The HSM's IPv4 address (X.Y.Z.K)
+	HSMIPAddress string `json:"hsmIpAddress"`
+
+	// +kubebuilder:validation:Optional
+	// The IP address of the client connecting to the HSM (X.Y.Z.K)
+	HSMClientAddress string `json:"hsmClientAddress"`
+
+        // +kubebuilder:validation:Optional
+        // The HSM certificates. The map's key is the OpenShift secret storing the certificate, and
+	// the value is the mounting point (e.g., "luna-certificates": "/usr/local/luna/config/certs").
+        HSMCertificates map[string]string `json:"hsmCertificates"`
```

Comment:
can there be multiple hsm certificates involved? because if I read the PR correct, only one would be mounted .  If if only one cert secret is used, can't we just mount them to a fixed patch, instead of making the path configurable?
If we need it to be configurable, why not have a list of a sub struct type with `secretName` and `mountPath` ?

Reply:
also what is the expected content of the secret? specific keys for referencing cert/key?

Reply:
This has been modified.  We can have multiple certs but they will all be contained  in a single secret - which will be mounted to a specific directory (mountpoint).  The contents of the secret will be all the files that will be created in that directory.